jdaviz/app.py

```diff
@@ -490,13 +495,25 @@ def _on_layers_changed(self, msg):
                 self.state.layer_icons = {**self.state.layer_icons,
                                           layer_name: orientation_icons.get(layer_name,
                                                                             wcs_only_refdata_icon)}
-            else:
+            elif is_not_child:
                 self.state.layer_icons = {
                     **self.state.layer_icons,
                     layer_name: alpha_index(len([ln for ln, ic in self.state.layer_icons.items()
-                                                 if not ic.startswith('mdi-')]))
+                                                 if not ic.startswith('mdi-') and
+                                                 self._get_assoc_data_parent(ln) is None]))
                 }
 
+        # all remaining layers at this point have a parent:
+        for layer_name in self.state.layer_icons:
+            children_layers = self._get_assoc_data_children(layer_name)
+            if children_layers is not None:
+                parent_icon = self.state.layer_icons[layer_name]
+                for i, child_layer in enumerate(children_layers, start=1):
+                    self.state.layer_icons = {
+                        **self.state.layer_icons,
+                        child_layer: f'{parent_icon}{i}'
+                    }
```

Comment:
this will update the state a bunch of times.  Can we instead either build this logic into above or use a dictionary-comprehension to build all the new entries and then just apply to the state once?

Reply:
Good idea. In 00785d7, I instead gather the child layer icon updates in a different dict, and update `self.state.layer_icons` once at the end.